GeneralsMD/Code/GameEngineDevice/Source/W3DDevice/GameClient/W3DView.cpp

```diff
@@ -226,10 +226,48 @@ void W3DView::setWidth(Int width)
  	m_3DCamera->Get_Viewport(vMin,vMax);
  	vMax.X=(Real)(m_originX+width)/(Real)TheDisplay->getWidth();
  	m_3DCamera->Set_Viewport(vMin,vMax);
+}
+
+// TheSuperHackers @tweak valeronm 25/03/2025 Adjusting FOV to look similar to what we have in in 4:3 (hFOV: 50, vFOV 38.55)
+void W3DView::adjustFovToAspectRatio(const Int width, const Int height)
+{
+	static const Real fixedHFOV = 50.00f;
+	static const Real fixedVFOV = 38.55f;
 
-	//we want to maintain the same scale, so we'll need to adjust the fov.
-	//default W3D fov for full-screen is 50 degrees.
-	m_3DCamera->Set_View_Plane((Real)width/(Real)TheDisplay->getWidth()*DEG_TO_RADF(50.0f),-1);
+	const Real aspectRatio = static_cast<Real>(width)/ static_cast<Real>(height);
+	Real vFOVRad, hFOVRad;
+	Bool drawEntireTerrain = false;
+	if (aspectRatio >= 4.0f / 3.0f)
```

Comment:
For Multiplayer, I very much prefer we cap this at max 16/9, min 9/16, to not give these ultra screens a massive advantage, preferably configurable in GameData.ini, so that Mods can do their own setups.

For Singleplayer and Replay it is fine to go wide with no limits.